include/mata/nfa.hh

```diff
@@ -1324,6 +1327,15 @@ public:
     std::list<Symbol> get_symbols() const override;
     std::list<Symbol> get_complement(const std::set<Symbol>& syms) const override;
 
+    std::string reverse_translate_symbol(const Symbol symbol) const override {
```

Comment:
Small suggestion: maybe this could have `default` parameter, that if set to some non-null value suppresses the throwing of the exception. I.e. when the default parameter is set by user, then default is returned instead of throwing exception. This could be useful for some, though it might be tricky to implement a little bit.

Reply:
What you want to use here is C++17 `std::optional`. This can be implemented in C++14 in various forms, but in this case, I would probably go with a pointer to the default value (pointer to `std::string`). The pointer is by default `nullptr`. If the user wants to return something else, they can pass an address of the desired string value. It is not a clean solution, but better than other approaches, I would say.

Furthermore, we can think about whether to add this functionality to the interface (abstract class `Alphabet`), hence force this feature on all subclasses, or to keep this extension only as `OnTheFlyAlphabet` specific function overload (with different name (e.g., `reverse_translate_symbol_with_default()`) to resolve function call ambiguity). I would be OK with both options. 

When we switch to C++17, we can switch to using `std::optional`.

Reply:
Yes, having two different functions, would be suitable as well. Maybe better than the solution with nullptr, which could be rather dirty. Maybe you can even overload it and we could implement the functionality in the Python similar to optional (since there, we have no type system, and hence we can consider None to call throwback and other value as calling with default).

Reply:
Yes, this can be done quite easily. The only question then is whether to force this behaviour on all alphabets (add the declaration to the interface class) or implement it only in the `OnTheFlyAlphabet`, as for example the `IntAlphabet` will not ever throw an exception for any input (it simply converts the passed symbol to string).

Reply:
We can leave this for discussion at next meeting.

Reply:
Will write it down to present this topic at our next meeting. I am sure everyone will be happy to hear we have another alphabet topic for them to discuss ;)

Reply:
Actaully, I think we don't need a default parameter at all. Do we have a use case for it? Or the solution with `nullptr` is ok for me. I'm against the solution with two functions. It complicates things. I don't think this topic is worth of discussion. If it become an issue in future, then we can invest time to it.

Reply:
Uses cases:
1. For display and debug: if there is some "unregistered" symbol, the exception will be only annoying.
2. One could use another (smaller) alphabet and treat unregistered symbol as `don't cares`, e.g. displaying `X` or some other.
3. For debugging, one could collect "badly registered symbols"

Yes, for "classic" algorithms, it is probably not useful, but for debugging and visualization it could be useful.

Reply:
Actually, these are not use cases. You need to debug or visualize _something_, in our case "classic" algorithms. Therefore I don't see it an actuall issue.

Reply:
Well, since Martin is clearly very passionate about this issue, and I can't seem to care as much and don't want to argue what is "use case" and what isn't (I disagree, what you wrote, display and debug is clear use case), I consider this issue closed.